src/riot-rs-runqueue/src/runqueue.rs

```diff
@@ -8,8 +8,23 @@ use self::clist::CList;
 const USIZE_BITS: usize = mem::size_of::<usize>() * 8;
 
 /// Runqueue number.
-pub type RunqueueId = u8;
-pub type ThreadId = u8;
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct RunqueueId(pub u8);
```

Comment:
> Would it make sense to make the inner value private instead, and provide a new() constructor, so that the value cannot be modified after the instance has been created?

Done in e77a204.

> Ideally, these newtypes would only be created in this module/crate, so that we could make the constructor private/pub(crate)-private, not sure whether this is possible/makes sense however.

That's unfortunately not possible. `riot-rs-threads` creates `RunqueueId`s based on the set thread priority, and `ThreadId`s are created based on availability. The logic for both is part of `riot-rs-threads`.